bot/exts/utilities/colour.py

```diff
@@ -0,0 +1,240 @@
+        """Create a dictionary mapping of colour types and their values."""
+        colour_name = self._rgb_to_name(rgb)
+        if colour_name is None:
+            colour_name = "No match found"
+        return {
+            "RGB": rgb,
+            "HSV": self._rgb_to_hsv(rgb),
+            "HSL": self._rgb_to_hsl(rgb),
+            "CMYK": self._rgb_to_cmyk(rgb),
+            "Hex": self._rgb_to_hex(rgb),
+            "Name": colour_name
+        }
+
+    @staticmethod
+    def _rgb_to_hsv(rgb: tuple[int, int, int]) -> tuple[int, int, int]:
+        """Convert RGB values to HSV values."""
+        rgb_list = [val / 255 for val in rgb]
+        h, s, v = colorsys.rgb_to_hsv(*rgb_list)
+        hsv = (round(h * 360), round(s * 100), round(v * 100))
+        return hsv
+
+    @staticmethod
+    def _rgb_to_hsl(rgb: tuple[int, int, int]) -> tuple[int, int, int]:
+        """Convert RGB values to HSL values."""
+        rgb_list = [val / 255.0 for val in rgb]
+        h, l, s = colorsys.rgb_to_hls(*rgb_list)
+        hsl = (round(h * 360), round(s * 100), round(l * 100))
+        return hsl
+
+    @staticmethod
+    def _rgb_to_cmyk(rgb: tuple[int, int, int]) -> tuple[int, int, int, int]:
+        """Convert RGB values to CMYK values."""
+        rgb_list = [val / 255.0 for val in rgb]
+        if not any(rgb_list):
+            return 0, 0, 0, 100
+        k = 1 - max(rgb_list)
+        c = round((1 - rgb_list[0] - k) * 100 / (1 - k))
+        m = round((1 - rgb_list[1] - k) * 100 / (1 - k))
+        y = round((1 - rgb_list[2] - k) * 100 / (1 - k))
+        cmyk = (c, m, y, round(k * 100))
+        return cmyk
+
+    @staticmethod
+    def _rgb_to_hex(rgb: tuple[int, int, int]) -> str:
+        """Convert RGB values to HEX code."""
+        hex_ = "".join([hex(val)[2:].zfill(2) for val in rgb])
+        hex_code = f"#{hex_}".upper()
+        return hex_code
+
+    def _rgb_to_name(self, rgb: tuple[int, int, int]) -> str:
+        """Convert RGB values to a fuzzy matched name."""
+        input_hex_colour = self._rgb_to_hex(rgb)
+        try:
+            match, certainty, _ = rapidfuzz.process.extractOne(
+                query=input_hex_colour,
+                choices=self.colour_mapping.values(),
+                score_cutoff=80
+            )
+            colour_name = [name for name, hex_code in self.colour_mapping.items() if hex_code == match][0]
+        except TypeError:
+            colour_name = None
+        return colour_name
+
+    def match_colour_name(self, input_colour_name: str) -> str:
+        """Convert a colour name to HEX code."""
+        try:
+            match, certainty, _ = rapidfuzz.process.extractOne(
+                query=input_colour_name,
+                choices=self.colour_mapping.keys(),
+                score_cutoff=80
+            )
+        except (ValueError, TypeError):
+            raise commands.BadArgument(message=f"No color found for: `{input_colour_name}`")
```

Comment:
Since the BadArgument is raised in the function `match_colour_name`, would this now need to be an async function so that it can return an embed directly?
```py
    def match_colour_name(self, input_colour_name: str) -> str:
        """Convert a colour name to HEX code."""
        try:
            match, certainty, _ = rapidfuzz.process.extractOne(
                query=input_colour_name,
                choices=self.colour_mapping.keys(),
                score_cutoff=80
            )
        except (ValueError, TypeError):
            raise commands.BadArgument(message=f"No color found for: `{input_colour_name}`")
        return f"#{self.colour_mapping[match]}"
```

Reply:
Made it async, here is what it looks like:

![image](https://user-images.githubusercontent.com/42116429/141210974-fcc0f3a9-3e6a-45bf-89bf-dd544798079e.png)